oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/oauth2-authorization-schema.sql

```diff
@@ -0,0 +1,27 @@
+CREATE TABLE oauth2_authorization (
+    id varchar(100) NOT NULL,
+    registered_client_id varchar(100) NOT NULL,
+    principal_name varchar(100) NOT NULL,
+    authorization_grant_type varchar(100) NOT NULL,
+    attributes varchar(1000) DEFAULT NULL,
+    state blob DEFAULT NULL,
+    authorization_code_value blob DEFAULT NULL,
+    authorization_code_issued_at timestamp DEFAULT NULL,
+    authorization_code_expires_at timestamp DEFAULT NULL,
+    authorization_code_metadata varchar(1000) DEFAULT NULL,
+    access_token_value blob DEFAULT NULL,
+    access_token_issued_at timestamp DEFAULT NULL,
+    access_token_expires_at timestamp DEFAULT NULL,
+    access_token_metadata varchar(1000) DEFAULT NULL,
+    access_token_type varchar(100) DEFAULT NULL,
+    access_token_scopes varchar(1000) DEFAULT NULL,
+    oidc_id_token_value blob DEFAULT NULL,
+    oidc_id_token_issued_at timestamp DEFAULT NULL,
+    oidc_id_token_expires_at timestamp DEFAULT NULL,
+    oidc_id_token_metadata varchar(1000) DEFAULT NULL,
+    refresh_token_value blob DEFAULT NULL,
+    refresh_token_issued_at timestamp DEFAULT NULL,
+    refresh_token_expires_at timestamp DEFAULT NULL,
+    refresh_token_metadata varchar(1000) DEFAULT NULL,
```

Comment:
For all code and token-related fields, this seems like a nice approach, given we want to focus on simplicity. The concern here would be if we ever need to add new token types, or realize we missed an attribute. Given that token types don't change often, and have a fairly fixed format, I think we can afford to stay flat here.

But I wanted to mention that it might be worth considering if there's any risk leaving it flat with all optional values. Pursuing a single-column JSON serialization scheme for tokens would be ideal for flexibility, but adds complexity and a more complicated API for this class. If you feel this ought to hold up well enough, I'll support leaving these columns as is.

Reply:
I was think about the JSON serialization solution as well in the beginning. But I decided to follow the same approach as in the `oauth2-client` module from `spring-security`. 

The other reason why I decided to use the flat format here is because we may need to search for the token without knowing the `tokenType`. And with this approach I can search for the token value in DB 
```
"state = ? OR authorization_code_value = ? OR access_token_value = ? OR refresh_token_value = ?"
```

If we serialize the token object as JSON, I think it will make it harder to search for a token value as above.